tests/lax_numpy_test.py

```diff
@@ -3873,14 +3873,21 @@ def testAstypeBool(self, from_dtype, use_method, to_dtype='bool'):
   @jtu.sample_product(
     change_dtype=[True, False],
     copy=[True, False],
+    change_device=[True, False],
   )
-  def testAstypeCopy(self, change_dtype, copy):
+  @jtu.run_on_devices("gpu")
+  def testAstypeCopy(self, change_dtype, copy, change_device):
     dtype = 'float32' if change_dtype else 'int32'
-    expect_copy = change_dtype or copy
+    device = jax.devices("cpu")[0] if change_device else None
+    expect_copy = change_dtype or copy or change_device
     x = jnp.arange(5, dtype='int32')
-    y = x.astype(dtype, copy=copy)
-
+    y = x.astype(dtype, copy=copy, device=device)
```

Comment:
Can you add a test for sharding too? Just testing single devices is not enough since you are adding sharding support too.

Please test all the codepaths you are adding support for :)